backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java

```diff
@@ -114,42 +114,44 @@ public Dataset createDataset(Dataset dataset, UserInfo userInfo)
       throws InvalidProtocolBufferException {
     // Check entity already exists
     checkDatasetAlreadyExist(dataset);
+    createRolesForDataset(dataset, userInfo);
 
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       Transaction transaction = session.beginTransaction();
       DatasetEntity datasetEntity = RdbmsUtils.generateDatasetEntity(dataset);
       session.save(datasetEntity);
-
-      Role ownerRole = roleService.getRoleByName(ModelDBConstants.ROLE_DATASET_OWNER, null);
-      roleService.createRoleBinding(
-          ownerRole,
-          new CollaboratorUser(authService, userInfo),
-          dataset.getId(),
-          ModelDBServiceResourceTypes.DATASET);
-      if (dataset.getDatasetVisibility().equals(DatasetVisibility.PUBLIC)) {
-        Role publicReadRole =
-            roleService.getRoleByName(ModelDBConstants.ROLE_DATASET_PUBLIC_READ, null);
-        UserInfo unsignedUser = authService.getUnsignedUser();
-        roleService.createRoleBinding(
-            publicReadRole,
-            new CollaboratorUser(authService, unsignedUser),
-            dataset.getId(),
-            ModelDBServiceResourceTypes.DATASET);
-      }
-
-      createWorkspaceRoleBinding(
-          dataset.getWorkspaceId(),
-          dataset.getWorkspaceType(),
-          dataset.getId(),
-          dataset.getDatasetVisibility());
-
       transaction.commit();
       LOGGER.debug("Dataset created successfully");
       TelemetryUtils.insertModelDBDeploymentInfo();
       return datasetEntity.getProtoObject();
     }
   }
 
+  private void createRolesForDataset(Dataset dataset, UserInfo userInfo) {
```

Comment:
createRoleBindingsForDataset  or createBindingsForDataset would be a better name, right?

Reply:
renamed